cpp/simulations/abm_braunschweig.cpp

```diff
@@ -0,0 +1,267 @@
+//  * The infection states are chosen randomly. They are distributed according to the probabilites set in the example.
+//  * @return random infection state
+//  */
+// mio::abm::InfectionState determine_infection_state(ScalarType exposed, ScalarType infected, ScalarType carrier,
+//                                                    ScalarType recovered)
+// {
+//     ScalarType susceptible          = 1 - exposed - infected - carrier - recovered;
+//     std::vector<ScalarType> weights = {susceptible,  exposed,      carrier,       infected / 3,
+//                                        infected / 3, infected / 3, recovered / 2, recovered / 2};
+//     if (weights.size() != (size_t)mio::abm::InfectionState::Count - 1) {
+//         mio::log_error("Initialization in ABM wrong, please correct vector length.");
+//     }
+//     auto state = mio::DiscreteDistribution<size_t>::get_instance()(weights);
+//     return (mio::abm::InfectionState)state;
+// }
+
+// /**
+//  * Assign an infection state to each person.
+//  */
+// void assign_infection_state(mio::abm::World& world, ScalarType exposed_pct, ScalarType infected_pct,
+//                             ScalarType carrier_pct, ScalarType recovered_pct)
+// {
+//     auto persons = world.get_persons();
+//     for (auto& person : persons) {
+//         world.set_infection_state(person,
+//                                   determine_infection_state(exposed_pct, infected_pct, carrier_pct, recovered_pct));
+//     }
+// }
+
+void split_line(std::string string, std::vector<int32_t>* row)
+{
+    std::vector<std::string> strings;
+    boost::replace_all(string, ";;", ";-1;");
+    boost::split(strings, string, boost::is_any_of(";"));
+    std::transform(strings.begin(), strings.end(), std::back_inserter(*row), [&](std::string s) {
+        return stoi(s);
+    });
+}
+
+mio::abm::LocationType get_location_type(uint32_t acitivity_end)
+{
+    mio::abm::LocationType type;
+    switch (acitivity_end) {
+    case 1:
+        type = mio::abm::LocationType::Work;
+        break;
+    case 2:
+        type = mio::abm::LocationType::School;
+        break;
+    case 3:
+        type = mio::abm::LocationType::BasicsShop;
+        break;
+    case 4:
+        type = mio::abm::LocationType::SocialEvent; // Freizeit
+        break;
+    case 5:
+        type = mio::abm::LocationType::Home; // Private Erledigung
+        break;
+    case 6:
+        type = mio::abm::LocationType::SocialEvent; // Sonstiges
+        break;
+    case 7:
+        type = mio::abm::LocationType::Home;
+        break;
+    }
+    return type;
+}
+
+void create_world_from_data(mio::abm::World& world, const std::string& filename)
+{
+    // File pointer
+    std::fstream fin;
+
+    // Open an existing file
+    fin.open(filename, std::ios::in);
+    std::vector<int32_t> row;
+    std::vector<std::string> row_string;
+    std::string line;
+
+    getline(fin, line);
```

Comment:
Please use std:: this looks like its a function you implemented above.

Reply:
changed